trace/workflow_execution_update_test.go

```diff
@@ -0,0 +1,263 @@
+package trace
+
+import (
+	"context"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"go.temporal.io/api/common/v1"
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/history/v1"
+	"go.temporal.io/api/workflow/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/mocks"
+)
+
+type WorkflowExecutionUpdateSuite struct {
+	suite.Suite
+	ctx context.Context
+}
+
+func TestWorkflowExecutionUpdateSuite(t *testing.T) {
+	suite.Run(t, new(WorkflowExecutionUpdateSuite))
+}
+
+func (s *WorkflowExecutionUpdateSuite) SetupTest() {
+	s.ctx = context.Background()
+}
+
+// SetDescribeWorkflowMocks mocks DescribeWorkflowExecution to return a list of events for a wfId/runId.
+// This is used in the WorkflowStateJob.Run method to retrieve the history length.
+func (s *WorkflowExecutionUpdateSuite) SetDescribeWorkflowMocks(client *mocks.Client, wfId, runId string, events []*history.HistoryEvent) {
```

Comment:
I'd prefer integration tests over unit tests with mocks to get better confidence that this code works with a real server.

Reply:
Good idea. Let me get to it 👍

Reply:
I've been giving this some thought and I'm not sure I understand what you're trying to convey. Is your suggestion to setup a server with fixture responses and test against that?

If that is the case: while I agree that a server would make these tests closer to real, I think leveraging the sdk mocks can make the tests easier to maintain (no server to maintain) and we can get exactly the response we want for each test. Additionally, failures in `DescribeWorkflowExecution` and `GetWorkflowHistory` in the sdk would bleed into the tests here.

In any case, I'm happy to review this and add more tests to better cover this functionality.

Reply:
My suggestion is that we test real server responses, that would mean starting a workflow and getting it to a specific state.

I think these unit tests are fine but I'd want at least a single integration test, can wait for when we implement the display logic.

Reply:
That's a significant investment into some tests which will probably not provide better coverage. If you believe it's going to add value, I invite you to implement it in the rest of the tool.

Reply:
Yes, we need more integration coverage in general but I won't block the PR on this.